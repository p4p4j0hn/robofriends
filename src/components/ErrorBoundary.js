import { Component } from 'react';

class ErrorBoundary extends Component {
  constructor() {
    super();
    this.state = {
      hasError: false
    }
  }

  componentDidCatch(error, info) {
    this.setState({ hasError: true });
  }

  render() {
    return this.state.hasError ?
      this.props.fallback
      :
      this.props.children;
  }
}

export default ErrorBoundary;
