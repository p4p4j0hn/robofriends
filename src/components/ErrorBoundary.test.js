import React from 'react';
import { shallow, mount } from 'enzyme';
import ErrorBoundary from './ErrorBoundary';
import util from 'util';

describe('errorboundary', () => {
  it('expect to render ErrorBoundary component', () => {
    expect.assertions(1);
    expect(shallow(<ErrorBoundary />)).toMatchSnapshot();
  });

  it('expect the component to catch', () => {
    expect.assertions(4);
    Object.defineProperty(global, 'TextEncoder', {
      value: util.TextEncoder
    });
    const MyComponent = () => <div>Render</div>;
    const DecoratedComponent = () => {
      return (
        <ErrorBoundary fallback={<div>Fallback</div>}>
          <MyComponent />
        </ErrorBoundary>
      );
    };
    const wrapper = mount(<DecoratedComponent />);
    expect(wrapper.contains(<div>Render</div>)).toBe(true);
    expect(wrapper.contains(<div>Fallback</div>)).toBe(false);
    wrapper.find(MyComponent).simulateError(new Error('Error While Rendering'));
    expect(wrapper.contains(<div>Render</div>)).toBe(false);
    expect(wrapper.contains(<div>Fallback</div>)).toBe(true);
  });
});
