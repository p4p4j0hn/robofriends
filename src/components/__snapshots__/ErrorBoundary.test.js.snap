// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`errorboundary expect to render ErrorBoundary component 1`] = `""`;
